src/detection/FootprintSet.cc

```diff
@@ -872,7 +872,7 @@ enum DIRECTION {
 template <typename MaskPixelT>
 class Startspan {
 public:
-    Startspan(geom::Span const *span, image::Mask<MaskPixelT> *mask, DIRECTION const dir);
+    Startspan(geom::Span const *span, image::Mask<MaskPixelT> *mask, DIRECTION dir);
```

Comment:
Is this actually discouraged in C++? I didn't know that. :thinking:

Reply:
It is not, but isn't part of the signature so possibly confuses people. I don't know what we prefer though, so we may not apply this check.

Reply:
I do occasionally find it useful to use a by-value const argument in order for the implementation to use the argument as a const variable.  And, of course, I dislike the fact that this is exposed in the signature.

So this feels like a situation where we should have a convention on whether this is allowed or not (I'm pretty ambivalent), and only do an automatic conversion if the convention discussion converges on prohibiting it.